.github/workflows/all-checks.yml

```diff
@@ -78,3 +78,39 @@ jobs:
         run: make install-test-requirements
       - name: Run linter for all starters
         run: behave features/lint.feature
+
+  package:
+    strategy:
+      matrix:
+        os: [ windows-latest , ubuntu-latest ]
+        python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12" ]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Set up Python ${{matrix.python-version}}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install test requirements
+        run: |
+          make install-test-requirements
+      - name: Add MSBuild to PATH
+        if: matrix.os == 'windows-latest'
+        uses: microsoft/setup-msbuild@v1
+      - name: Install Microsoft Visual C++ Redistributable
+        if: matrix.os == 'windows-latest'
+        run: |
+          choco install vcredist-all
+      - name: Setup Hadoop binary
+        if: matrix.os == 'windows-latest'
+        run: |
+          Invoke-WebRequest "https://github.com/steveloughran/winutils/blob/master/hadoop-2.7.1/bin/winutils.exe?raw=true" -OutFile winutils.exe
+          Invoke-WebRequest "https://github.com/steveloughran/winutils/blob/master/hadoop-2.7.1/bin/hadoop.dll?raw=true" -OutFile hadoop.dll
+          Move-Item .\hadoop.dll C:\Windows\System32
+          New-Item -ItemType directory -Path C:\hadoop\bin
+          Move-Item .\winutils.exe C:\hadoop\bin
+          echo "HADOOP_HOME=C:\hadoop" | Out-File -Append -Encoding ascii -FilePath $env:GITHUB_ENV
+          echo "PATH=$env:HADOOP_HOME\bin;$env:PATH" | Out-File -Append -Encoding ascii -FilePath $env:GITHUB_ENV
```

Comment:
Do we need this for starter? I am actually surprised we bring back Hadoop setup into `kedro` @ankatiyar, why do we need this? As I understand we don't have dataset settings, so I don't expect we need hadoop other than `kedro-datasets`

Reply:
Without Hadoop setup spark starters were failing on window: https://github.com/kedro-org/kedro-starters/actions/runs/10705672782/job/29685497693

Reply:
Yeah for the pyspark starters ^